### PR TITLE
[candi] wait node before annotate node

### LINKED ISSUE #6057: [bashible] when joining a cluster, wait for nodes to be created in the cluster
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Version

v1.52.0

### Expected Behavior

wait until the nodes object is added and then try to add the annotation afterwards 

### Actual Behavior

Now when adding a node an error is thrown for the first time when trying to add an annotation

### Steps To Reproduce

_No response_

### Additional Information

https://flant.slack.com/archives/CML93AWG2/p1696422148883149

### Logs

```shell
=== Step: /var/lib/bashible/bundle_steps/098_remove_old_binaries.sh
===
===
=== Step: /var/lib/bashible/bundle_steps/099_reboot.sh
===
Error from server (NotFound): nodes "d8-worker-01" not found
Failed to annotate node d8-worker-01 with annotation node.deckhouse.io/configuration-checksum=1a70a8d65e313a67e50812691167a61679101b455faa86d0a9a215fc0783cdef ... retry in 10 seconds.
```

### PR DESCRIPTION
**Pull request by artbin**
## Description

Add a bash function that waits for a node to be added to the cluster before annotating the node.

## Why do we need it, and what problem does it solve?

Fix #6057.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: candi
type: fix
summary: Wait for a node to be added to the cluster before annotating the node.
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/openstack'...

**Comment by artbin** [comment]
Bootstrap log:
```
===
=== Step: /var/lib/bashible/bundle_steps/099_reboot.sh
===
===
=== Step: /var/lib/bashible/bundle_steps/100_disable-ntp-on-node.sh
===
Removed /etc/systemd/system/sysinit.target.wants/systemd-timesyncd.service.
Error from server (NotFound): nodes "akulyabin-main-ubuntu-18-c87545be-dd7f9-x7mrk" not found
Failed to check existence of node akulyabin-main-ubuntu-18-c87545be-dd7f9-x7mrk ... retry in 10 seconds.
```

**Review by RomanenkoDenys — APPROVED** [review]
LGTM

**Comment by artbin** [comment]
/e2e/abort openstack;Standard;containerd;1.25 7081652941 7081652941-1-con-1-25 /sys/deckhouse-oss/install:pr6443 debian@5.159.103.10

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/openstack'...

**Comment by artbin** [comment]
/e2e/abort openstack;Standard;containerd;1.25 7081709233 7081709233-1-con-1-25 /sys/deckhouse-oss/install:pr6443 debian@185.91.54.3

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @RomanenkoDenys. I've started the workflow for label 'e2e/run/aws'...

File: candi/bashible/bashible.sh.tpl [hunk 1]
```diff
@@ -55,7 +55,26 @@ function bb-event-error-create() {
 EOF
 }
 
+function wait_node() {
+  attempt=0
+  error=$(kubectl_exec get node $(hostname -s) 2>&1 >/dev/null)
```
Comment (RomanenkoDenys) [inline candi/bashible/bashible.sh.tpl:60]: kubectl_exec executes only one time.

**Review by RomanenkoDenys — CHANGES_REQUESTED** [review]
some comments

File: candi/bashible/bashible.sh.tpl (hunk 1 above)
Comment (artbin) [reply candi/bashible/bashible.sh.tpl]: fixed

**Review by RomanenkoDenys — APPROVED** [review]

**Review by RomanenkoDenys — APPROVED** [review]
LGTM

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/aws'...

**Comment by name212** [comment]
/e2e/abort aws;WithoutNAT;containerd;1.25 7478011868 7478011868-1-con-1-25 /sys/deckhouse-oss/install:pr6443 centos@18.194.132.209

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @artbin. I've started the workflow for label 'e2e/run/aws'...

**Review by name212 — APPROVED** [review]

**Pull request merged**